MitoMathHelper.js

```diff
@@ -4,82 +4,143 @@
             return tickPercentage;
-        }else{
-            return null;
         }
+        return null;
+    }
 
+    /**
+     * returns distance ^ 2
+     * @param pointA
+     * @param pointB
+     * @returns {number}
+     * @private
+     */
+    static _distanceBetweenTwoPointsNoSqrt(pointA, pointB) {
+        let xDifference = pointB[0] - pointA[0];
+        let yDifference = pointB[1] - pointA[1];
+        return xDifference * xDifference + yDifference * yDifference;
     }
 
-    distanceBetweenTwoPoints(pointA, pointB){
+    /**
+     * returns distance between two points
+     * @param pointA
+     * @param pointB
+     * @returns {number}
+     * @private
+     */
+    static _distanceBetweenTwoPoints(pointA, pointB) {
         return Math.hypot(pointB[0] - pointA[0], pointB[1] - pointA[1]);
     }
 
-    detectCollisionCirCir(circleA, circleB){
+    /**
+     * detects if two circles overlap
+     * @param pointA
+     * @param radiusA
+     * @param pointB
+     * @param radiusB
+     * @returns {boolean}
+     * @private
+     */
+    static _detectCollision(pointA, radiusA, pointB, radiusB) {
         let distance = Math.hypot(
-            circleB.getPosition()[0] - circleA.getPosition()[0],
-            circleB.getPosition()[1] - circleA.getPosition()[1]);
-        let radiusTotal = circleA.getRadius() + circleB.getRadius();
-        return  radiusTotal >= distance;
+            pointA[0] - pointB[0],
+            pointA[1] - pointB[1]);
+        let radiusTotal = radiusA + radiusB;
+        return radiusTotal >= distance;
     }
 
-    getClosestPoint(linePointA, linePointB, singlePoint){
-        let A_to_P = [singlePoint[0] - linePointA[0], singlePoint[1] - linePointA[1]] ;
-        let A_to_B = [linePointB[0] - linePointA[0], linePointB[1] - linePointA[1]];
-        let atb2 = (A_to_B[0] * A_to_B[0]) + (A_to_B[1] * A_to_B[1]);
-
-        if(atb2 === 0){return linePointA;}
-
-        let atp_dot_atb = (A_to_P[0]*A_to_B[0]) + (A_to_P[1] * A_to_B[1]);
-        let normalizedDistance = atp_dot_atb / atb2;
-        return [linePointA[0] + (A_to_B[0] * normalizedDistance) , linePointA[1] + (A_to_B[1] * normalizedDistance)]
+    /**
+     * return the closest point on a line to a single point
+     * @param linePointA
+     * @param linePointB
+     * @param singlePoint
+     * @returns {*}
+     * @private
+     */
+    static _getClosestPoint(linePointA, linePointB, singlePoint) {
+        let aToP = [singlePoint[0] - linePointA[0], singlePoint[1] - linePointA[1]];
+        let aToB = [linePointB[0] - linePointA[0], linePointB[1] - linePointA[1]];
+        let aToB2 = (aToB[0] * aToB[0]) + (aToB[1] * aToB[1]);
+        if (aToB2 === 0) {
+            return linePointA;
+        }
+        let atpDotAtb = (aToP[0] * aToB[0]) + (aToP[1] * aToB[1]);
+        let normalizedDistance = atpDotAtb / aToB2;
+        return [linePointA[0] + (aToB[0] * normalizedDistance), linePointA[1] + (aToB[1] * normalizedDistance)]
```

Comment:
you dont need () around the multiplication here but it doesnt really matter either way